workspaces/diff-engine/src/projections/spec_events.rs

```diff
@@ -45,7 +46,7 @@ impl SpecAssemblerProjection {
       .root_chunk
       .ok_or_else(|| SpecAssemblerError::RootChunkRequired)?;
 
-    let mut current_chunk_id = root_chunk.id.clone();
+    let mut current_chunk_id = root_chunk.last_batch_id().clone();
```

Comment:
Batch commits from outside chunks can append to the "specification.json" by referring to the last batch commit it contains.